relay-server/src/processing/transactions/mod.rs

```diff
@@ -375,27 +450,98 @@ impl<T: Counted + AsRef<Annotated<Event>>> RateLimited for Managed<ExpandedTrans
     {
         let scoping = self.scoping();
 
+        // These debug assertions are here because this function does not check indexed or extracted
+        // span limits.
+        // TODO: encode flags into types instead.
+        debug_assert!(!self.flags.metrics_extracted);
+        debug_assert!(!self.flags.spans_extracted);
+
         let ExpandedTransaction {
             headers: _,
-            transaction,
+            event,
             flags,
             attachments,
             profile,
             extracted_spans,
+            category: _,
         } = self.as_ref();
 
         // If there is a transaction limit, drop everything.
         // This also affects profiles that lost their transaction due to sampling.
-        for (category, quantity) in transaction.quantities() {
+        let limits = rate_limiter
+            .try_consume(scoping.item(DataCategory::Transaction), 1)
+            .await;
```

Comment:
If the transaction is limited, we can already drop the entire payload and don't need to consider attachments and profiles, right?